src/core/common/CustomLayout.js

```diff
@@ -50,4 +54,11 @@ const CustomLayout = ({ classes, ...rest }) => {
     );
 };
 
-export default withStyles(styles)(CustomLayout);
+const mapStateToProps = state => {
+    const isContrastMode = get(state, 'custom.contrastMode.data', false);
+    return {
+        theme: getCurrentTheme(isContrastMode),
```

Comment:
warning: `getCurrentTheme` returns a new object each time you call it... this will harm the application performance big time.

Reply:
Yes, it is understandable. But I don't know other way to provide possibility to toggle contrast mode. The main parameter of contrast mode is black color instead of theme color

Reply:
use lodash's memoize.

https://lodash.com/docs/4.17.11#memoize